crates/astria-sequencer/src/service/mempool.rs

```diff
@@ -74,7 +74,18 @@ fn handle_check_tx(req: request::CheckTx) -> response::CheckTx {
             };
         }
     };
-    let signed_tx = SignedTransaction::try_from_raw(raw_signed_tx).unwrap();
+    let signed_tx = match SignedTransaction::try_from_raw(raw_signed_tx) {
+        Ok(tx) => tx,
+        Err(e) => {
+            return response::CheckTx {
+                code: AbciCode::INVALID_SIGNATURE.into(),
+                info: "failed to verify signed transaction".into(),
```

Comment:
This is set from the abci code description.

I believe thats the intended use of the codes.

More context is then provided through the logs

Reply:
updated to return the error as well